support/cas-server-support-saml-idp-metadata/src/main/java/org/apereo/cas/support/saml/services/idp/metadata/cache/resolver/MetadataQueryProtocolMetadataResolver.java

```diff
@@ -72,38 +80,38 @@ protected HttpResponse fetchMetadata(final String metadataLocation, final Criter
         return response;
     }
 
-    /**
-     * Is dynamic metadata query configured ?
-     *
-     * @param service the service
-     * @return true/false
-     */
-    protected boolean isDynamicMetadataQueryConfigured(final SamlRegisteredService service) {
-        return service.getMetadataLocation().trim().endsWith("/entities/{0}");
-    }
-
     @Override
     public boolean supports(final SamlRegisteredService service) {
-        return isDynamicMetadataQueryConfigured(service);
+        return SamlUtils.isDynamicMetadataQueryConfigured(service.getMetadataLocation());
     }
 
     @Override
     protected boolean shouldHttpResponseStatusBeProcessed(final HttpStatus status) {
-        return super.shouldHttpResponseStatusBeProcessed(status) || status == HttpStatus.NOT_MODIFIED;
+        return true;
     }
 
     @Override
     protected AbstractMetadataResolver getMetadataResolverFromResponse(final HttpResponse response, final File backupFile) throws Exception {
-        if (response.getStatusLine().getStatusCode() == HttpStatus.NOT_MODIFIED.value()) {
-            return new InMemoryResourceMetadataResolver(backupFile, this.configBean);
+        if (!HttpStatus.valueOf(response.getStatusLine().getStatusCode()).is2xxSuccessful()) {
+            if (Files.exists(backupFile.toPath())) {
+                return new InMemoryResourceMetadataResolver(backupFile, this.configBean);
```

Comment:
What happens if the backup file as expired metadata?

Reply:
With this implementation we just fail over and use it, instead of not loading anything.  Is it desirable or possibly even necessary that we don't allow expired metadata to be used?

Reply:
Yes, and now that I double check, I think that bit is handled by OpenSAML and what you're doing here is consistent with all other resolvers. Expiration handling of metadata is done at resolution time, not at retrieval time so we're good here.